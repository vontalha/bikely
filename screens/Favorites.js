import React, {useEffect, useState, useLayoutEffect, PermissionsAndroid,} from 'react';
import {View, Text, FlatList, Button, Image, StyleSheet, ScrollView} from 'react-native';
import {auth, db} from '../config/Firebase';
import {collection, doc, getDocs, getDoc} from 'firebase/firestore';
import BikeCardMedium from '../components/BikeCardMedium';

const FavoritenScreen = ({navigation}) => {
  const [favoriten, setFavoriten] = useState([]);
  const [locationData, setLocationData] = React.useState(null);

function extend(obj, src) {
  for (var key in src) {
      if (src.hasOwnProperty(key)) obj[key] = src[key];
  }
  return obj;
}

useLayoutEffect(() => {
  const loadFavorites = async () => {
    const favoritenData = [];
    if (auth.currentUser != null) {
      const userSnap = await getDoc(doc(db, 'users', auth.currentUser.uid));

      const favoritenIDs = userSnap.data().favourites;

      const querySnapshot = await getDocs(collection(db, 'listings'));
      querySnapshot.forEach(doc => {
        if (favoritenIDs.includes(doc.id)) {

          favoritenData.push(extend(doc.data(),{_id: doc.id}));
        }
      });
    }
    setFavoriten(favoritenData);
  };

  loadFavorites();
}, [setFavoriten]);

  
  return (
    <ScrollView style={styles.scrollView}>
      <Text>Favoriten</Text>
      {auth.currentUser != null ? (
        favoriten.map(favorite => <BikeCardMedium key={favorite._id} listing={favorite} navigation={navigation} />)
      ) : (
        <View>
          <Text>Your saved bikes live here.</Text>
          <Text>
            Once you start saving bikes you like, they will appear here.
          </Text>
          <Button
            title="Discover Bikes"
            onPress={() => navigation.navigate('Home')}
          />
        </View>
      )}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    marginHorizontal: 16,
  },
});

export default FavoritenScreen;
