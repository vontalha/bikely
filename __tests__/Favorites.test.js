import React from 'react';
import renderer from 'react-test-renderer';
import Favorites from '../screens/Favorites';

describe('Favorites', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Favorites />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
