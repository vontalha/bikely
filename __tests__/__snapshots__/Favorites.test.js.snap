// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites renders correctly 1`] = `
<RCTScrollView>
  <View>
    <Text>
      Favoriten
    </Text>
    <View>
      <Text>
        Your saved bikes live here.
      </Text>
      <Text>
        Once you start saving bikes you like, they will appear here.
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontWeight": "500",
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          DISCOVER BIKES
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
